### ISSUE TITLE
Poor results when using previous binary segmentation

### ISSUE THREAD

**Issue by joshmyersdean** [facebookresearch/segment-anything#447]
Hello!

When I try to input a previous binary segmentation into SAM I observe poor results compared to using logits. Is there any way to perform corrections on a pre-existing binary segmentation?

**Comment by etiennedub** [comment]
Hi!

I'm also interested in editing an existing mask with SAM. I did try to generate the logits from a grid of clicks based on a binary mask but it's doesn't really work. The initial logits is ok but it's not really editable from subsequent clicks:

![screenshot-2023-06-21_13:42:46](https://github.com/facebookresearch/segment-anything/assets/14828908/38d1749b-d598-4af3-bc2e-8c47d92c9d83)

TOP: Grid of clicks determined by a binary mask (left) and the resulting logits (right)
BOTTOM: Logits from previous step and negative click (left) and the resulting logits (right)

The result is a really low confidence logits that is basiclly non-usable. I got similar result with diffrent images. 

If you find a way to edit a mask from SAM, feel free to share!
